packages/legacy/core/App/types/navigators.ts

```diff
@@ -209,4 +210,9 @@ export type HistoryStackParams = {
   [Screens.HistoryPage]: undefined
 }
 
-export type ScreenOptionsType = Partial<Record<Screens, StackNavigationOptions>>
+export type StackNavigationOptionsExtended = StackNavigationOptions & {
+  screenEdges?: Edge[] | undefined
+  customHeader?: () => React.ReactNode | undefined
+}
+
+export type ScreenOptionsType = Partial<Record<Screens, StackNavigationOptionsExtended>>
```

Comment:
ScreenOptionsDictionary is used in the OnboardingStack to pass options to the screen (see below). With this change in type, the screen options can now include screenEdges and customHeader, but these two properties should not be passed to the screen options.
<img width="626" alt="image" src="https://github.com/user-attachments/assets/fe95158f-0b63-4c94-925f-d4a984ff2e2c">

Reply:
Yes, it shouldn't be passed to the screen options, and if you try to pass them, listing will fail. but there is no harm from using the abstract type "ScreenOptionsType" to include them for the sack of using them in the wrapper.

Reply:
Yes, but the ScreenOptionsDictionary has already been passed to the screen options in the onboardingStack. The token in the useService was specifically created for this purpose, so I'm not entirely sure I understand what you're referring to.

Reply:
It shouldn't cause any issues, but could be a concern. I will take another look at the architecture

Reply:
done